test/propagation_test.cpp

```diff
@@ -92,3 +92,26 @@ TEST_CASE("SpanContext") {
     }
   }
 }
+
+TEST_CASE("Binary Span Context") {
+  std::stringstream carrier{};
```

Comment:
If I were mean, I'd make you write tests for each of the cases where it can fail too...

Reply:
I added a few more.